### PR TITLE
[PXD-2145] Use new config everywhere

### PR DESCRIPTION
**Pull request by roofridge**
- Use new config everywhere* instead of Flask app config
- *Except for in fence/__init__.py for backwards compatibility 
- Change (simplify) multi-tenant tests, partially to facilitate above

### Deployment changes

### CODE REVIEW

**Comment by PlanXCyborg** [comment]
This PR contains code that is not formatted correctly according to [`black`](https://github.com/ambv/black). Run `black` on your code before merging.

<details>
<summary>Expand the full diff to see formatting changes</summary>

```diff
--- fence/auth.py
+++ blackened
@@ -119,17 +119,13 @@
                     flask.request, flask.session["username"], flask.session["provider"]
                 )
                 return f(*args, **kwargs)
 
             eppn = None
-            enable_shib = "shibboleth" in config.get(
-                "ENABLED_IDENTITY_PROVIDERS", []
-            )
+            enable_shib = "shibboleth" in config.get("ENABLED_IDENTITY_PROVIDERS", [])
             if enable_shib and "SHIBBOLETH_HEADER" in config:
-                eppn = flask.request.headers.get(
-                    config["SHIBBOLETH_HEADER"]
-                )
+                eppn = flask.request.headers.get(config["SHIBBOLETH_HEADER"])
 
             if config.get("MOCK_AUTH") is True:
                 eppn = "test"
             # if there is authorization header for oauth
             if "Authorization" in flask.request.headers:
@@ -155,11 +151,11 @@
     if flask.session.get("username"):
         login_user(flask.request, flask.session["username"], flask.session["provider"])
 
     eppn = flask.request.headers.get(config["SHIBBOLETH_HEADER"])
 
-    if config.get("MOCK_AUTH") is True:  
+    if config.get("MOCK_AUTH") is True:
         eppn = "test"
     # if there is authorization header for oauth
     if "Authorization" in flask.request.headers:
         has_oauth(scope=scope)
     # if there is shibboleth session, then create user session and

--- fence/blueprints/data.py
+++ blackened
@@ -124,14 +124,11 @@
             raise NotFound("Can't find any file locations.")
 
         return signed_url
 
     def _get_index_document(self):
-        indexd_server = (
-            config.get("INDEXD")
-            or config["BASE_URL"] + "/index"
-        )
+        indexd_server = config.get("INDEXD") or config["BASE_URL"] + "/index"
         url = indexd_server + "/index/"
         try:
             res = requests.get(url + self.file_id)
         except Exception as e:
             flask.current_app.logger.error(
@@ -275,13 +272,11 @@
             ),
         }
 
     def get_credential_to_access_bucket(self, aws_creds, expires_in):
         s3_buckets = get_value(
-            config,
-            "S3_BUCKETS",
-            InternalError("buckets not configured"),
+            config, "S3_BUCKETS", InternalError("buckets not configured")
         )
         if len(aws_creds) == 0 and len(s3_buckets) == 0:
             raise InternalError("no bucket is configured")
         if len(aws_creds) == 0 and len(s3_buckets) > 0:
             raise InternalError("credential for buckets is not configured")
@@ -311,23 +306,23 @@
                 aws_creds, bucket_cred, cred_key, expires_in
             )
 
     def get_signed_url(self, action, expires_in, public_data=False):
         aws_creds = get_value(
-            config,
-            "AWS_CREDENTIALS",
-            InternalError("credentials not configured"),
+            config, "AWS_CREDENTIALS", InternalError("credentials not configured")
         )
 
         http_url = "https://{}.s3.amazonaws.com/{}".format(
             self.parsed_url.netloc, self.parsed_url.path.strip("/")
         )
 
         credential = self.get_credential_to_access_bucket(aws_creds, expires_in)
 
         aws_access_key_id = get_value(
-            credential, "aws_access_key_id", InternalError("aws configuration not found")
+            credential,
+            "aws_access_key_id",
+            InternalError("aws configuration not found"),
         )
         if aws_access_key_id == "*":
             return http_url
 
         region = flask.current_app.boto.get_bucket_region(

--- fence/blueprints/login/google.py
+++ blackened
@@ -3,10 +3,11 @@
 
 from fence.auth import login_user
 from fence.errors import UserError
 from fence.models import IdentityProvider
 from fence.config import config
+
 
 class GoogleRedirect(Resource):
     def get(self):
         flask.redirect_url = flask.request.args.get("redirect")
         if flask.redirect_url:

--- fence/blueprints/login/shib.py
+++ blackened
@@ -31,13 +31,11 @@
         """
         Complete the shibboleth login.
         """
 
         if "SHIBBOLETH_HEADER" in config:
-            eppn = flask.request.headers.get(
-                config["SHIBBOLETH_HEADER"]
-            )
+            eppn = flask.request.headers.get(config["SHIBBOLETH_HEADER"])
 
         else:
             raise InternalError("Missing shibboleth header configuration")
         username = eppn.split("!")[-1] if eppn else None
         if username:

--- fence/blueprints/oauth2.py
+++ blackened
@@ -69,17 +69,13 @@
         user = get_current_user()
     except Unauthorized:
         need_authentication = True
 
     if need_authentication or not user:
-        redirect_url = (
-            config.get("BASE_URL") + flask.request.full_path
-        )
+        redirect_url = config.get("BASE_URL") + flask.request.full_path
         params = {"redirect": redirect_url}
-        login_url = add_params_to_uri(
-            config.get("DEFAULT_LOGIN_URL"), params
-        )
+        login_url = add_params_to_uri(config.get("DEFAULT_LOGIN_URL"), params)
         return flask.redirect(login_url)
 
     try:
         grant = server.validate_consent_request(end_user=user)
         grant.validate_prompt(user)

--- fence/error_handler.py
+++ blackened
@@ -5,11 +5,11 @@
 from flask import render_template
 from authlib.specs.rfc6749.errors import OAuth2Error
 from werkzeug.exceptions import HTTPException
 
 from fence.errors import APIError
-from fence.config import config 
+from fence.config import config
 
 
 def get_error_response(error):
     details, status_code = get_error_details_and_status(error)
     support_email = config.get("SUPPORT_EMAIL_FOR_ERRORS")

--- fence/resources/google/utils.py
+++ blackened
@@ -138,13 +138,11 @@
     private_key_bytes = json.dumps(sa_private_key).encode("utf-8")
     private_key = fernet_key.encrypt(private_key_bytes)
 
     expires = expires or (
         int(time.time())
-        + config[
-            "GOOGLE_SERVICE_ACCOUNT_KEY_FOR_URL_SIGNING_EXPIRES_IN"
-        ]
+        + config["GOOGLE_SERVICE_ACCOUNT_KEY_FOR_URL_SIGNING_EXPIRES_IN"]
     )
 
     add_custom_service_account_key_expiration(
         key_id, service_account.id, expires, private_key=private_key
     )
@@ -652,13 +650,13 @@
     main/default application credentials as the monitoring service account.
 
     This function should ONLY return the service account's email by
     parsing the creds file.
     """
-    app_creds_file = app_creds_file or config.get(
-        "CIRRUS_CFG", {}
-    ).get("GOOGLE_APPLICATION_CREDENTIALS")
+    app_creds_file = app_creds_file or config.get("CIRRUS_CFG", {}).get(
+        "GOOGLE_APPLICATION_CREDENTIALS"
+    )
 
     creds_email = None
     if app_creds_file and os.path.exists(app_creds_file):
         with open(app_creds_file) as app_creds_file:
             creds_email = json.load(app_creds_file).get("client_email")

--- fence/resources/user/user_session.py
+++ blackened
@@ -120,13 +120,13 @@
 
     def clear_if_expired(self, app):
         if self._encoded_token:
             now = int(time.time())
             is_expired = self.session_token["exp"] <= now
-            end_of_life = self.session_token["context"][
-                "session_started"
-            ] + config.get("SESSION_LIFETIME")
+            end_of_life = self.session_token["context"]["session_started"] + config.get(
+                "SESSION_LIFETIME"
+            )
             lifetime_over = end_of_life <= now
             if is_expired or lifetime_over:
                 self.clear()
         else:
             # if there's no current token set, clear data to be sure

--- tests/google/test_service_accounts.py
+++ blackened
@@ -118,13 +118,11 @@
 
     mock_path = patch("fence.blueprints.google.os", path_mock)
     # mock_path = patch('os.path.exists', True)
     mocked_open = patch("__builtin__.open", mock_open(read_data=creds_file))
 
-    monkeypatch.setitem(
-        config, "CIRRUS_CFG", {"GOOGLE_APPLICATION_CREDENTIALS": "."}
-    )
+    monkeypatch.setitem(config, "CIRRUS_CFG", {"GOOGLE_APPLICATION_CREDENTIALS": "."})
 
     mocked_open.start()
     mock_path.start()
     response = client.get(
         "/google/service_accounts/monitor",

--- tests/login/login_shib.py
+++ blackened
@@ -1,24 +1,21 @@
 from fence.config import config
+
 
 def test_shib_redirect(client, app):
     r = client.get("/login/shib")
     assert r.status_code == 302
 
 
 def test_shib_login(app, client):
-    r = client.get(
-        "/login/shib/login", headers={config["SHIBBOLETH_HEADER"]: "test"}
-    )
+    r = client.get("/login/shib/login", headers={config["SHIBBOLETH_HEADER"]: "test"})
     assert r.status_code == 200
 
 
 def test_shib_login_redirect(app, client):
     r = client.get("/login/shib?redirect=http://localhost")
-    r = client.get(
-        "/login/shib/login", headers={config["SHIBBOLETH_HEADER"]: "test"}
-    )
+    r = client.get("/login/shib/login", headers={config["SHIBBOLETH_HEADER"]: "test"})
     assert r.status_code == 302
     assert r.headers["Location"] == "http://localhost"
 
 
 def test_shib_login_fail(client):

--- tests/login/test_default_login.py
+++ blackened
@@ -1,6 +1,7 @@
 from fence.config import config
+
 
 def test_default_login(app, client):
     response_json = client.get("/login").json
     assert "default_provider" in response_json
     assert "providers" in response_json
```
</details></br>

This formatting comment was generated automatically by a script in [uc-cdis/wool](https://github.com/uc-cdis/wool).

File: tests/conftest.py [hunk 1]
```diff
@@ -1059,34 +1057,13 @@ def google_storage_client_mocker(app):
 
 
 @pytest.fixture(scope="function")
-def remove_google_idp(app):
+def config_fixture():
```
Comment (Avantol13) [inline tests/conftest.py]: minor nitpick:  I wouldn't have `fixture` in the name, maybe call it `restore_config` or something?

File: tests/oidc/core/authorization/test_id_token_hint.py [hunk 2]
```diff
@@ -30,10 +30,11 @@
 
 from fence.jwt.validate import validate_jwt
 from tests.utils import oauth2
+from fence.config import config
 
 
 @pytest.mark.skip(
-    reason="We are NOT COMPLAINT for this OPTIONAL param (id_token_hint) yet."
+    reason="We are NOT COMPLIANT for this OPTIONAL param (id_token_hint) yet."
```
Comment (Avantol13) [inline tests/oidc/core/authorization/test_id_token_hint.py]: lol, thanks for fixing these. "we are not complaint" is really funny tho

**Comment by Avantol13** [comment]
I'd still have @rudyardrichter or @philloooo verify that the multi-tenant tests make sense though

File: tests/conftest.py (hunk 1 above)
Comment (roofridge) [reply tests/conftest.py]: 👌 👌

File: tests/oidc/core/authorization/test_id_token_hint.py (hunk 2 above)
Comment (roofridge) [reply tests/oidc/core/authorization/test_id_token_hint.py]: Indeed. I almost didn't complain... 👀

**Pull request merged**